### PR TITLE
Rework shutdown

### PR DESCRIPTION
**Pull request by osheroff**
1 - make sure we shutdown maxwell before we shutdown log4j.  I think this was the source of some deadlocks.
2 - add a connectTimeout on the jdbc connection so we don't spin endlessly.

@zendesk/rules 

https://github.com/zendesk/maxwell/issues/441

### CODE REVIEW

**Comment by dadah89** [comment]
b/c the error came from socketConnect I thought it'd need to socket timeout, but afaict if that was the case then the exception would be `SocketTimeoutException`, so I guess it's fine

File: src/main/java/com/zendesk/maxwell/MaxwellMysqlConfig.java [hunk 1]
```diff
@@ -16,13 +16,18 @@
 	public String user;
 	public String password;
 	public ArrayList<String> jdbcOptions = new ArrayList<String>();
+	public Integer connectTimeoutMS = 5000;
```
Comment (dadah89) [inline src/main/java/com/zendesk/maxwell/MaxwellMysqlConfig.java]: `final`, and all caps if you care about java const name conventions

**Comment by dadah89** [comment]
just a comment about `final`, besides looks good to me 👍

File: src/main/java/com/zendesk/maxwell/MaxwellMysqlConfig.java (hunk 1 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellMysqlConfig.java]: nah, figured it'd be loosely configurable, at least in code

**Pull request merged**